dotnet/src/SemanticKernel.UnitTests/Functions/KernelFunctionFromMethodTests1.cs

```diff
@@ -1104,6 +1104,94 @@ public async Task ItPropagatesOriginalExceptionTypeAsync()
         Assert.Same(expected, actual);
     }
 
+    public class ComplexType
+    {
+        public int Foo { get; set; }
+    }
+
+    [Fact]
+    public async Task ItSupportsMarshallingStringToComplexTypeAsync()
+    {
+        //Arrange
+        static ComplexType Test(ComplexType complexType) => complexType;
+
+        // Act
+        var function = KernelFunctionFactory.CreateFromMethod(Method(Test));
+
+        FunctionResult result = await function.InvokeAsync(this._kernel, arguments: new() { { "complexType", @"{ ""Foo"": 42 }" } });
+        Assert.NotNull(result);
+
+        var complexTypeResult = result.GetValue<ComplexType>();
+        Assert.NotNull(complexTypeResult);
+        Assert.Equal(42, complexTypeResult.Foo);
+    }
+
+    [Fact]
+    public async Task ItSupportsMarshallingComplexTypeToStringAsync()
+    {
+        //Arrange
+        static ComplexType Test(ComplexType complexType) => complexType;
+
+        // Act
+        var function = KernelFunctionFactory.CreateFromMethod(Method(Test));
+
+        FunctionResult result = await function.InvokeAsync(this._kernel, arguments: new() { { "complexType", new ComplexType() { Foo = 42 } } });
+        Assert.NotNull(result);
+
+        var stringResult = result.ToString();
+        Assert.NotNull(stringResult);
+        Assert.Equal(@"{""Foo"":42}", stringResult);
+    }
+
+    [Fact]
+    public async Task ItThrowsWhenMarshallingComplexTypeToNonStringTypeAsync()
+    {
+        //Arrange
+        static int Test(int complexType) => complexType;
+
+        // Act
+        var function = KernelFunctionFactory.CreateFromMethod(Method(Test));
+
+        int count = 0;
+        try
+        {
+            await function.InvokeAsync(this._kernel, arguments: new() { { "complexType", new ComplexType() { Foo = 42 } } });
+            Assert.Fail("Should have thrown an ArgumentException since complex type cannot be marshalled to int");
+        }
+        catch (ArgumentException)
+        {
+            count++;
+        }
+
+        Assert.Equal(1, count);
```

Comment:
All these lines can be replaced by this one - await Assert.ThrowsAsync<ArgumentException>(() => fawait function.InvokeAsync(this._kernel, arguments: new() { { "complexType", new ComplexType() { Foo = 42 } } }));